### DISCUSSION TITLE
Difference between "with LogixDriver('ip') as plc", and plc = LogixDriver('ip'), pycomm3.exceptions.CommError: A session must be registered before a Forward Open

### DISCUSSION THREAD

**Discussion by jds8086** [ottowayi/pycomm3#223] [Q&A]
Can someone explain to me the functional difference between 

```
with LogixDriver('10.52.44.10') as plc:
    print(plc.read('BoilerOneFeedWaterPressure').value)
``` 

and

```
plc = LogixDriver('10.52.44.10')
print(plc.read('BoilerOneFeedWaterPressure').value)
```

The first works just fine, the second throws the exception: 

```
pycomm3.exceptions.CommError: A session must be registered before a Forward Open
```

**Comment by pcwii** [comment]
You have a typo in the second one. You're are missing the opening quote for the ip address.  Cheers!

**Reply by jds8086** [reply]
Thanks, odd that it's there in my ide, so this isn't the issue.

I guess i should add that the root of the issue is that i do not want to use 'with' and have to connect to the plc every time i want to read a tag or set of tags, mainly because of the delay when initially connecting. I want to connect once, then do reads as necessary from there. All of the examples i see are using 'with'

**Reply by jds8086** [reply]
Found .open() in another users post here, it's pretty obvious i guess but i missed that bit. It's working as i wanted it too now (or at least i'm past that roadblock, time to start the real fun)

```
plc = LogixDriver('10.52.44.10')
plc.open()
print(plc.read('BoilerOneFeedWaterPressure').value)
```
